tree/dataframe/src/RDFColumnRegister.cxx

```diff
@@ -23,7 +23,7 @@ void RColumnRegister::AddColumn(const std::shared_ptr<RDFDetail::RDefineBase> &c
    auto newDefines = std::make_shared<RDefineBasePtrMap_t>(GetColumns());
    const std::string colName(name);
    (*newDefines)[colName] = column;
-   fDefines = newDefines;
+   fDefines = std::move(newDefines);
```

Comment:
Why don't you assign the result of `make_shared` to `fDefines` directly?

Reply:
good point, I guess there was a reason at some point -- will fix

Reply:
ok, found out again: `fDefines` is a `shared_ptr<const T>`, so we have to modify the contents of `newDefines` before assigning them to a pointer to const.

Reply:
That probably deserves a in-code comment :)